dm/master/scheduler/latch.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
+
+import (
+	"sync"
+
+	"github.com/pingcap/errors"
+)
+
+// latches provides a simple way to block concurrent accessing to one resource, if caller tries to acquire latch before
+// accessing resources.
+type latches struct {
+	mu    sync.Mutex
+	inUse map[string]struct{}
```

Comment:
can we consider using some third party implementation? such as [kmutex](https://github.com/im7mortal/kmutex)

Reply:
the API is slightly different, kmutex will be blocked which will slow the response to user. Let's wait other reviewers' opinion.

Also we can fork or contribute to that repo for a TryLock method.

Reply:
seems we can't use a library of GPL-3.0 License